tokio/src/runtime/driver/op.rs

```diff
@@ -8,8 +9,11 @@ use std::task::Poll;
 use std::task::Waker;
 use std::{io, mem};
 
+#[allow(dead_code)]
 #[derive(Debug)]
-pub(crate) enum CancelData {}
+pub(crate) enum CancelData {
+    Open(Open),
+}
```

Comment:
Why the dead code annotation?

Reply:
I removed some `#[allow(dead_code)]` annotations, but it seems that some of them still require the annotation. So I also added comments to explain that.

[1405ee8](https://github.com/tokio-rs/tokio/pull/7321/commits/1405ee87b11c6bbb558d2f411002fd587524cd88)